src/query_parser/parser.y

```diff
@@ -843,11 +834,16 @@ vector_query(A) ::= vector_command(B). {
 
 // Every vector query will have basic command part. Right now we only have TOP_K command.
 // It is this rule's job to create the new vector node for the query.
-vector_command(A) ::= TOP_K param_size(B) modifier(C) ATTRIBUTE(D). {
-  D.type = QT_PARAM_VEC;
-  A = NewVectorNode_WithParams(ctx, VECSIM_QT_TOPK, &B, &D);
-  A->vn.vq->property = rm_strndup(C.s, C.len);
-  RedisModule_Assert(-1 != (rm_asprintf(&A->vn.vq->scoreField, "__%.*s_score", C.len, C.s)));
+vector_command(A) ::= TERM(T) param_size(B) modifier(C) ATTRIBUTE(D). {
+  if (!strncmp("TOP_K", T.s, T.len)) {
+    D.type = QT_PARAM_VEC;
+    A = NewVectorNode_WithParams(ctx, VECSIM_QT_TOPK, &B, &D);
+    A->vn.vq->property = rm_strndup(C.s, C.len);
+    RedisModule_Assert(-1 != (rm_asprintf(&A->vn.vq->scoreField, "__%.*s_score", C.len, C.s)));
+  } else {
+    reportSyntaxError(ctx->status, &T, "Syntax error: Expecting Vector Similarity command");
```

Comment:
Is there a test with illegal `TOP_K`?

Reply:
Yes, both in the unit test and test_vecsim.py